protocol/message_handler.go

```diff
@@ -722,3 +731,32 @@ func (m *MessageHandler) HandleEmojiReaction(state *ReceivedMessageState, pbEmoj
 
 	return nil
 }
+
+func (m *MessageHandler) HandleGroupChatInvitation(state *ReceivedMessageState, pbGHInvitations protobuf.GroupChatInvitation) error {
+	logger := m.logger.With(zap.String("site", "HandleGroupChatInvitation"))
+	if err := ValidateReceivedGroupChatInvitation(&pbGHInvitations); err != nil {
+		logger.Error("invalid group chat invitation", zap.Error(err))
+		return err
+	}
+
+	groupChatInvitation := &GroupChatInvitation{
+		GroupChatInvitation: pbGHInvitations,
+		SigPubKey:           state.CurrentMessageState.PublicKey,
+	}
+
+	if groupChatInvitation.Rejection {
+		groupChatInvitation.From = types.EncodeHex(crypto.FromECDSAPub(&m.identity.PublicKey))
```

Comment:
I am not sure I understand this,
why do we set `From` to us if it's a rejection?
That would change the ID of the invitation

Reply:
From is always request author, so i use the same invitation instance and change its state , but from it always the same and means the initial author of the request, probably i need to rename the field

Reply:
yes i was not sure about this code as well, i feel like something wrong here but not sure how to improve that

Reply:
The `&m.identity.PublicKey` refers to the identity of the current user, in the [rejection sending code](https://github.com/status-im/status-go/pull/2013/files/6c5e2f8fef9de45d64496e698ac81febbad2d4b2#r476370135) you set the `From` field to the original requester.

Did you find that the returning rejection message had a different `From` key than you expected?